ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,146 @@
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
+
+    def __get_authselect_current_output(self):
+        current = ipautil.run(
+            [
+                paths.AUTHSELECT,
+                "current"
+            ],
+            env={"LC_ALL": "C.UTF8"}
+        )
+
+        if current.returncode != 0:
+            return None
+
+        return current.raw_output.decode()
+
+    def __parse_authselect_output(self, output_text=None):
+        if output_text is None:
+            output_text = self.__get_authselect_current_output()
+
+        if output_text is None:
+            return None
+
+        cfg_params = re.findall(
+            r"\s*Profile ID:\s*(\S+)\s*\n\s*Enabled features:\s*(.*)",
+            output_text,
+            re.DOTALL
+        )
+
+        profile = cfg_params[0][0]
+
+        if not profile:
+            return None
+
+        features = re.findall("-\s*(\S+)", cfg_params[0][1], re.DOTALL)
+
+        return (profile, features)
 
-class RedHatAuthConfig(object):
+    def configure(self, sssd, mkhomedir, statestore):
+        if not sssd:
+            logger.error("no-sssd options is not supported by installer. "
+                         "For client installation without sssd check the "
+                         "output of 'ipa-advise config-fedora-authconfig'")
+            return
+            # raise ScriptError("Non-sssd installation is not supported")
+
+        cfg = self.__parse_authselect_output()
+        if cfg:
+            statestore.backup_state('authselect', 'profile', cfg[0])
+            statestore.backup_state('authselect', 'features_list',
+                                    " ".join(cfg[1]))
+
+        cmd = [paths.AUTHSELECT, "select", "sssd"]
+        if mkhomedir:
+            cmd.append("with-mkhomedir")
+        cmd.append("--force")
+
+        ipautil.run(cmd)
+
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+
+        if not statestore.has_state('authselect'):
+            return
+
+        profile = statestore.restore_state('authselect', 'profile')
+        features = statestore.restore_state('authselect', 'features_list')
+
+        cmd = [paths.AUTHSELECT, "select", profile, features, "--force"]
+        ipautil.run(cmd)
+
+    def backup(self, path):
+        current = ipautil.run([
+                paths.AUTHSELECT,
+                "current"
+        ])
+
+        with open(os.path.join(path, "authselect.backup"), 'wb') as f:
```

Comment:
This will trigger an error as /var/lib/ipa/auth_backup does not exist. The error does not happen with authconfig because authconfig --savebackup creates the directory.

Reply:
Good catch. Thank you.